libc-bottom-half/libpreopen/libpreopen.c

```diff
@@ -155,6 +156,23 @@ stat(const char *path, struct stat *st)
     return fstatat(dirfd, relative_path, st, AT_SYMLINK_NOFOLLOW);
 }
 
+int
+utime(const char *path, const struct utimbuf *times)
+{
+    const char *relative_path;
+    int fd = __wasilibc_find_relpath(path, &relative_path);
+
+    // If we can't find a preopened directory handle to open this file with,
+    // indicate that the program lacks the capabilities.
+    if (fd == -1) {
+        errno = ENOTCAPABLE;
+        return -1;
+    }
+    return utimensat(fd, relative_path, times ? ((struct timespec [2]){
+		{ .tv_sec = times->actime }, { .tv_sec = times->modtime }})
+        : 0,  AT_SYMLINK_NOFOLLOW);
```

Comment:
Is this intended to be `AT_SYMLINK_NOFOLLOW`? I don't have a definitive reference handy, but most other things in POSIX follow symlinks.

Reply:
I've followed lead with the above `int stat` function https://github.com/WebAssembly/wasi-libc/blob/36e6c09f998ea233a1f57b1e4322e1ca9ab222ac/libc-bottom-half/libpreopen/libpreopen.c#L156

Reply:
Ah, that's an error, likely an accidental copy+pasto from `lstat`. It should be 0.